netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -263,7 +272,7 @@ public void operationComplete(ChannelFuture future) throws Exception {
       }
     });
     // Start the connection operation to the server.
-    channel.connect(address);
+    channel.connect(remoteAddress);
```

Comment:
This is surprising.  From what I can tell, the socket is created at channel creation time, not at connect, so I assumed the local address already had to be present.  

I didn't test this myself, since our CI doesn't have a stable address to test this with.  @normanmaurer can comment if this is expected behavior.

Reply:
Yes - it is surprising, but it appears that localAddress only takes effect on a bind. Not sure we want to be doing a bind on the channel (though internally, that is what is probably getting called on the socket). 

I tried this locally by building the library with this diff and it works for me:

ngshetty-Mac:grpc-java ngshetty$ git diff
diff --git a/netty/src/main/java/io/grpc/netty/NettyClientTransport.java b/netty/src/main/java/io/grpc/netty/NettyClientTransport.java
index 2be71c20d..7cf964b58 100644
--- a/netty/src/main/java/io/grpc/netty/NettyClientTransport.java
+++ b/netty/src/main/java/io/grpc/netty/NettyClientTransport.java
@@ -272,7 +272,11 @@ class NettyClientTransport implements ConnectionClientTransport {
       }
     });
     // Start the connection operation to the server.
-    channel.connect(remoteAddress);
+    if (localAddress == null) {
+      channel.connect(remoteAddress);
+    } else {
+      channel.connect(remoteAddress, localAddress);
+    }

     if (keepAliveManager != null) {
       keepAliveManager.onTransportStarted();